api.md

```diff
@@ -287,6 +288,12 @@ in the table.
 to how you handle constraints. A hypertable can contain foreign keys to normal SQL table
 columns, but the reverse is not allowed. UNIQUE and PRIMARY constraints must include the
 partitioning key.
+>
+>The deadlock is likely to happen when concurrent transactions simultaneously try 
+to insert data into tables that are referenced in the foreign key constraints and into the
+converting table itself. The deadlock can be prevented by manually obtaining `SHARE ROW EXCLUSIVE` lock 
+on the referenced tables before calling `create_hypertable` in the same transaction,
+see [PostgreSQL documentation][postgres-lock] for the syntax.
```

Comment:
```suggestion
>If you have a foreign key constraint from the hypertable to a normal SQL table, a deadlock can occur if concurrent transactions try to insert data into both the referenced table and the hypertable. The deadlock can be prevented by manually locking the referenced table using `SHARE ROW EXCLUSIVE` in the transaction before calling `create_hypertable`. See [PostgreSQL documentation][postgres-lock] for the syntax.
```

Reply:
@mkindahl Thank you for providing suggestion. I see several issues with it, which make it more complicated and not improving over the current text:
1. There is no hypertable at this moment, so I don't think it is correct to say `hypertable`.
2. The beginning of the suggestion repeats what is already said in the beginning of the warning. I think it is important to avoid repeating text. The beginning will be suitable if the paragraph is moved into a separate warning as it was originally. The issue is that it will be too many warnings and many reviewers think it should not be a warning, however I don't think it is a tip either.
3. I tried to fix your last sentence to be correct, but then it becomes more ambiguous.

I am not fully satisfied with my text, but I wasn't able to find a way to really improve it. I think I already spent too much time on this text.